src/main/java/fi/vrk/xroad/monitor/actor/Supervisor.java

```diff
@@ -0,0 +1,122 @@
+package fi.vrk.xroad.monitor.actor;
+
+import akka.actor.AbstractActor;
+import akka.actor.ActorRef;
+import akka.actor.OneForOneStrategy;
+import akka.actor.SupervisorStrategy;
+import akka.japi.pf.DeciderBuilder;
+import akka.pattern.Patterns;
+import akka.routing.SmallestMailboxPool;
+import akka.util.Timeout;
+import fi.vrk.xroad.monitor.extensions.SpringExtension;
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import scala.concurrent.Await;
+import scala.concurrent.duration.Duration;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static akka.actor.SupervisorStrategy.resume;
+import static fi.vrk.xroad.monitor.util.MonitorCollectorConstants.SUPERVISOR_MONITOR_DATA_ACTOR_POOL_SIZE;
+
+/**
+ * Supervisor for actors
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class Supervisor extends AbstractActor {
+
+  @Getter
+  private ActorRef resultCollectorActor;
+  private ActorRef monitorDataRequestPoolRouter;
+
+  @Autowired
+  SpringExtension ext;
+
+  public Supervisor (ActorRef resultCollectorActor, ActorRef monitorDataRequestPoolRouter) {
+    this.resultCollectorActor = resultCollectorActor;
+    this.monitorDataRequestPoolRouter = monitorDataRequestPoolRouter;
+  }
+
+  public Supervisor () {}
+
+  @Override
+  public void preStart() throws Exception {
+    log.info("preStart");
+
+    if (resultCollectorActor == null) {
```

Comment:
This conditional initialization (one for tests and one for actual use) is a bit iffy. One problem is that actor hierarchy which is created for tests is different from actual (e.g. resultCollectorActor is not a child of Supervisor, and supervisorStrategy has no effect to it's error handling). I will research a bit, but if there's no good alternative, at least add good comments to avoid confusion.